src/auth.ts

```diff
@@ -158,15 +151,66 @@ async function authorizeWithoutLocalhost() {
  * @param {boolean} options.localhost authorize without http server.
  * @param {string} options.creds location of credentials file.
  */
-export function login(options: { localhost: boolean, creds: string}) {
-  DOTFILE.RC.read().then((rc: ClaspSettings) => {
-    console.warn(ERROR.LOGGED_IN);
-  }).catch(async (err: string) => {
-    DOTFILE.RC_LOCAL.read().then((rc: ClaspSettings) => {
-      console.warn(ERROR.LOGGED_IN);
-    }).catch(async (err: string) => {
-      await checkIfOnline();
-      authorize(options.localhost, options.creds);
-    });
-  });
+export async function login(options: { localhost: boolean, creds: string }) {
+  if ((options.creds && localOathSettingsExist()) ||
```

Comment:
I think this would be more readable:
```ts
const hasLocalCreds = ...;
const hasLocalCreds = ...;
if (hasLocalCreds || hasLocalCreds) {
  ...
}
```